src/main/java/nextstep/subway/path/domain/PathFinder.java

```diff
@@ -0,0 +1,41 @@
+package nextstep.subway.path.domain;
+
+import nextstep.subway.path.exception.SourceEqualsWithTargetException;
+import nextstep.subway.path.exception.StationNotExistsException;
+import nextstep.subway.path.exception.StationsNotConnectedException;
+import nextstep.subway.station.domain.Station;
+import org.jgrapht.GraphPath;
+import org.jgrapht.alg.shortestpath.DijkstraShortestPath;
+import org.jgrapht.graph.DefaultWeightedEdge;
+import org.jgrapht.graph.WeightedMultigraph;
+
+import java.util.Optional;
+
+public class PathFinder {
+    private WeightedMultigraph<Station, DefaultWeightedEdge> graph;
+
+    public PathFinder(WeightedMultigraph<Station, DefaultWeightedEdge> graph) {
+        this.graph = graph;
+    }
+
+    public GraphPath getShortestPath(Station source, Station target) {
+        if(source.equals(target)) {
+            throw new SourceEqualsWithTargetException();
+        }
+
+        DijkstraShortestPath dijkstraShortestPath = new DijkstraShortestPath(graph);
```

Comment:
```suggestion
        DijkstraShortestPath<Station, DefaultWeightedEdge> dijkstraShortestPath = new DijkstraShortestPath<>(graph);
```

generic은 항상 명시화해주면 좋답니다 👍